### ISSUE TITLE
Do not delete frozen accounts.

### ISSUE THREAD

**Issue by mikhailUshakoff** [tonlabs/ton-labs-node#210]
The storage size of a frozen account is minimal, about 1 cell and 327 bits, as it only stores the hash of the stateInit. Currently, accounts are deleted when their debt(due) exceeds 1 ever. However, if an account is frozen with a balance near 0, the time before it is deleted is more than 30 years.

For safety reasons, we propose not to delete frozen accounts. This will ensure that the account remains in the blockchain and can be unfrozen and used again in the future if needed.

**Comment by Rexagon** [comment]
The problem with account deletion is that the previously large contract could accumulate a fairly large debt, and now, when it goes into a frozen state, this debt is preserved and the subsequent transaction deletes it.

I agree with the proposed solution, since it is not so expensive to store frozen accounts.

**Comment by mnill** [comment]
I agree, this is an absolutely logical change. As @rexagon said, the main issue with deletion is that a contract can accumulate a very large debt before it gets frozen, and then it will be removed immediately without a 3-30 year waiting period. However, my wish is that this should be marked as temporary, meaning that currently we are not deleting contracts but only freezing them. But in the future, when, for example, storage transactions are introduced, which will prevent contracts from accumulating a large debt and ensure they are frozen in time, we will revert back to deletion. This is because if you write contracts using opcodes, it's very cheap to create many contracts, and they do have some impact since it's necessary to update the merkle proof of the shard's state.

**Comment by rualekseev** [comment]
Commit ID: eb074537d868d9feb5acffb3dbb60f55dd2c7982
Node Version: 0.59.10
Block version: 56


https://github.com/everx-labs/ever-node/pull/258/ Support capability CapUndeletableAccounts
